_StylizedAOVAdaptor : Add RenderMan NPR AOVs automatically

johnhaddon opened the pull request:
We trigger this based on the existence of one of the stylised display filters in the scene. This automates the fiddliest part of the setup when using Stylized Looks (and the only part that requires no artistic input). I went through several iterations on this, with the final version being the simplest of them all. I started by adding the minimal subset required by the specific filters present, since not all filters require all AOVs. Then I discovered that some of the filters depend on different AOVs based on their parameter settings, and it seemed rather over-complex to extend the scene analysis that far. For any realistic purposes you're going to end up needing most of the AOVs anyway, and its a decent reduction in complexity just to request them all unconditionally. Likewise, I made no effort to deduplicate the outputs declared by this and the _InteractiveDenoiserAdaptor. The RenderMan backend will deduplicate to a single `riley::RenderOutput` where relevant anyway, and the `null` display driver presumably has very little overhead.

murraystevenson reviewed (commented):
Thanks John! This does look like a very nice quality of life improvement for anyone working with stylized looks. I've noted a couple of minor things inline, feel free to squash and rebase as you go.

johnhaddon commented on python/GafferRenderMan/_StylizedAOVAdaptor.py:
```diff
@@ -0,0 +1,133 @@
+			"NPRNtriplanar", "NPRPtriplanar", "NPRalbedo", "NPRcurvature", "NPRcurvature", "NPRdistort",
+			"NPRdistort", "NPRhatchOut", "NPRlineAlbedo", "NPRlineCamdist", "NPRmask", "NPRlineNZ",
+			"NPRlineOut", "NPRlineOutAlpha", "NPRlineWidth", "NPRoutline", "NPRsections",
+			"NPRshadow", "NPRtextureCoords", "NPRlineNZ", "NPRtoonOut", "Nn", "Palbedo"
```
Doh. Sorry for the sloppiness. I started out with separate lists of requirements for each display filter, and then botched the merging of them into one.

johnhaddon commented on python/GafferRenderMan/_StylizedAOVAdaptor.py:
```diff
@@ -0,0 +1,133 @@
+			"NPRNtriplanar", "NPRPtriplanar", "NPRalbedo", "NPRcurvature", "NPRcurvature", "NPRdistort",
+			"NPRdistort", "NPRhatchOut", "NPRlineAlbedo", "NPRlineCamdist", "NPRmask", "NPRlineNZ",
+			"NPRlineOut", "NPRlineOutAlpha", "NPRlineWidth", "NPRoutline", "NPRsections",
+			"NPRshadow", "NPRtextureCoords", "NPRlineNZ", "NPRtoonOut", "Nn", "Palbedo"
```
Good catch - I've updated that and done a render with hatched shadows to confirm that it is working.

johnhaddon commented on python/GafferRenderMan/_StylizedAOVAdaptor.py:
```diff
@@ -0,0 +1,133 @@
+			"NPRNtriplanar", "NPRPtriplanar", "NPRalbedo", "NPRcurvature", "NPRcurvature", "NPRdistort",
+			"NPRdistort", "NPRhatchOut", "NPRlineAlbedo", "NPRlineCamdist", "NPRmask", "NPRlineNZ",
+			"NPRlineOut", "NPRlineOutAlpha", "NPRlineWidth", "NPRoutline", "NPRsections",
+			"NPRshadow", "NPRtextureCoords", "NPRlineNZ", "NPRtoonOut", "Nn", "Palbedo"
```
I think it is what happens when a muppet concatenates two separate AOV names.

johnhaddon commented:
> I've noted a couple of minor things inline, feel free to squash and rebase as you go.

Thanks for catching those - I've addressed all comments and squashed into https://github.com/GafferHQ/gaffer/pull/6360/commits/9b8b2b39065065abe414aede0ae27e35ec909e7f.
